blockchain/peer_test.go

```diff
@@ -0,0 +1,216 @@
+package blockchain
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	cmn "github.com/tendermint/tendermint/libs/common"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/p2p"
+)
+
+// These vars are used to record and verify different events generated during tests
+var (
+	numErrFuncCalls int        // number of calls to the errFunc
```

Comment:
avoid mutable global state. This should probably be a part of `bpTestPeer` or something